hetmech/hetmat.py

```diff
@@ -39,6 +40,32 @@ def hetmat_from_graph(graph, path):
     return hetmat
 
 
+def hetmat_from_permuted_graph(hetmat, permutation_id, permuted_graph):
+    """
+    Assumes subdirectory structure and that permutations inherit nodes but not
+    edges.
+    """
+    if '.' in permutation_id:
+        raise ValueError('Permutations ids cannot contain `.`')
```

Comment:
Is this a problem because directories cannot have a period in their name? Are ids usually the names like `hetionet-v1.0-perm-1`?

Reply:
periods now allowed by https://github.com/greenelab/hetmech/pull/100/commits/ca006c046eef8295a044b8a2bb24ffa27958f984. The implementation is not much uglier, so the ability to support `.` makes sense.